Shared/Tests.Shared/AddOrUpdateTests.cs

```diff
@@ -813,6 +813,39 @@ public void Add_WhenRequiredPropertyIsNotSet_ShouldThrow()
             }, Throws.TypeOf<RealmException>());
         }
 
+        [Test]
+        public void AddOrUpdate_WhenObjectHasNonEmptyList_ShouldNotThrow()
+        {
+            var first = new PrimaryKeyWithPKList
+            {
+                Id = 42,
+                StringValue = "value1"
+            };
+
+            first.ListValue.Add(new PrimaryKeyObject
+            {
+                Id = 1
+            });
+
+            _realm.Write(() => _realm.Add(first));
+
+            var second = new PrimaryKeyWithPKList
+            {
+                Id = 42,
+                StringValue = "value2"
+            };
+
+            second.ListValue.Add(new PrimaryKeyObject
```

Comment:
What is the expected fate of this object, what effect does it have on what is stored in the Realm. I find this test a bit confusing.